### PR TITLE
Feature video listing

### PR DESCRIPTION
**Pull request by OmKakatkar**
## Features
* Updated Video DB
* Dynamic Video Card
* Dynamic Video Grid
* Video Context with error handling
* Styles for Header and Sidebar
image

### CODE REVIEW

File: src/frontend/components/VideoCard/VideoCard.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import ReactPlayer from 'react-player/youtube';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faEllipsisVertical } from '@fortawesome/free-solid-svg-icons';
+import { getImageUrl, getVideoUrl } from '../../utils/video-helpers';
+import './VideoCard.css';
+
+export function VideoCard({ videoData }) {
+	const { _id, title, category } = videoData;
+	console.table(videoData);
```
Comment (manish591) [inline src/frontend/components/VideoCard/VideoCard.js]: Remove console if not required

File: src/frontend/containers/VideoContainer/VideoContainer.css [hunk 2]
```diff
@@ -0,0 +1,10 @@
+.video-grid {
+  margin-left: var(--width-sidebar);
+  margin-top: var(--height-header);
+	display: grid;
+	grid-template-columns: repeat(auto-fill, minmax(23rem, 1fr));
```
Comment (manish591) [inline src/frontend/containers/VideoContainer/VideoContainer.css:4]: Fix indentation

File: src/frontend/shared/NavBar/Header.css [hunk 3]
```diff
@@ -0,0 +1,10 @@
+.header {
+	position: fixed;
+  width: 100%;
```
Comment (manish591) [inline src/frontend/shared/NavBar/Header.css]: fix indentation here also

**Review by manish591 — APPROVED** [review]
The code is looking good, good naming convention and the code is modular too.

In some places, your code is not properly indented. I would suggest you to use prettier for consistent formatting

**Comment by OmKakatkar** [comment]
> The code is looking good, good naming convention and the code is modular too.
> 
> In some places, your code is not properly indented. I would suggest you to use prettier for consistent formatting

Thanks for the review.
I have prettier installed, but it doesn't seem to be working properly today. Will fix all indentation issues.

**Review by surajmore98 — APPROVED** [review]
good naming convention.
clean and well structured code.

**Review by Logan1x — APPROVED** [review]
Looks good to proceed. The `state` of the `video context` has all the data. maybe make it a object and have child element as `videoData` in future :)

File: src/frontend/utils/video-helpers.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export const getVideoUrl = id => {
+	return `https://youtu.be/${id}`;
```
Comment (praveenreddy1798) [inline src/frontend/utils/video-helpers.js]: No need to use return here, You can use it like this => **export const getVideoUrl = id => `https://youtu.be/${id}`**

File: src/frontend/utils/video-helpers.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+};
+
+export const getImageUrl = id => {
+	return `https://i3.ytimg.com/vi/${id}/sddefault.jpg`;
```
Comment (praveenreddy1798) [inline src/frontend/utils/video-helpers.js]: No need to use return here, you can do like this **export const getImageUrl = id => `https://i3.ytimg.com/vi/${id}/sddefault.jpg`;**

File: src/frontend/context/video-context.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+				...state,
+				videoList: action.payload
+			};
+		default:
+			return state;
+	}
+};
+
+const VideoProvider = ({ children }) => {
+	const [state, dispatch] = useReducer(reducer, { videoList: [] });
```
Comment (praveenreddy1798) [inline src/frontend/context/video-context.js]: you can use better naming like videosState and videosDispatch

File: src/frontend/components/VideoCard/VideoCard.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+import { getImageUrl } from '../../utils/video-helpers';
+import './VideoCard.css';
+
+export function VideoCard({ videoData }) {
+	const { _id, title, category } = videoData;
+	return (
+		<article className="video-card">
+			<Link to={`/watch/${category}/${_id}`}>
+				<div className="card-media">
+					<img src={getImageUrl(_id)} alt={title} />
```
Comment (praveenreddy1798) [inline src/frontend/components/VideoCard/VideoCard.js:13]: can use lazy loading for images

File: src/frontend/utils/video-helpers.js (hunk 4 above)
Comment (OmKakatkar) [reply src/frontend/utils/video-helpers.js]: Yes, will make the changes

File: src/frontend/utils/video-helpers.js (hunk 5 above)
Comment (OmKakatkar) [reply src/frontend/utils/video-helpers.js]: Yes, will make the changes

File: src/frontend/context/video-context.js (hunk 6 above)
Comment (OmKakatkar) [reply src/frontend/context/video-context.js]: Yes, will make the changes

File: src/frontend/components/VideoCard/VideoCard.js (hunk 7 above)
Comment (OmKakatkar) [reply src/frontend/components/VideoCard/VideoCard.js:13]: Currently there is no much data in the db, so the images will most likely be on the view port on Page load. Will add a lazy loading component in the future

**Pull request merged**